scene/resources/physics_material.cpp

```diff
@@ -43,9 +43,9 @@ void PhysicsMaterial::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_absorbent", "absorbent"), &PhysicsMaterial::set_absorbent);
 	ClassDB::bind_method(D_METHOD("is_absorbent"), &PhysicsMaterial::is_absorbent);
 
-	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "friction"), "set_friction", "get_friction");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "friction", PROPERTY_HINT_RANGE, "0,1,0.01"), "set_friction", "get_friction");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "rough"), "set_rough", "is_rough");
-	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "bounce"), "set_bounce", "get_bounce");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "bounce", PROPERTY_HINT_RANGE, "0,1,0.01"), "set_bounce", "get_bounce");
```

Comment:
You can use `or_greater` hint while still restricting the slider to `0..1` range, which should be common for most use cases anyway, but still allow for specific use cases (for values greater than >1, can be manually typed via keyboard), see #36450 for instance.


```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "bounce", PROPERTY_HINT_RANGE, "0.0,1.0,0.01,or_greater"), "set_bounce", "get_bounce");
```

Reply:
Interestingly, the new behavior in `master` is that you can still drag the slider past the max value, but the slider will be snapped to the specified range instead, this is not the case for Godot 3.2, IIRC.

Reply:
Good to know! Given that > 1 didn't seem to be meaningful here after all, I think hard limits make sense.

Reply:
Hmm I guess I just happen to have some physics bodies in 2D with custom physics, it's possible to set values above >1 just like Akien mentioned, so in theory the physics material could be used for non-engine classes, since it's agnostic to the physics backend used. It's a small nuance, but that's something which I'd do myself, but the engine development philosophy is that everything must be kept simple to cater the most common use cases, so I can only comply. 😛

Reply:
No, it doesn't:

https://github.com/godotengine/godot/blob/9a463cb1d62a174aa34f608c0281963cdad255e9/scene/resources/physics_material.cpp#L62-L65

Reply:
Makes sense. Maybe `or_greater` would be appropriate here then. I'll leave that to you to PR if you think its useful though :)

Reply:
The physics engines do clamp the calculated bounce to 0..1:
```
body_pair_2d_sw.cpp
214:real_t combine_bounce(Body2DSW *A, Body2DSW *B) {
215:    return CLAMP(A->get_bounce() + B->get_bounce(), 0, 1);
```
(at least in 2D)

Sure you can set a negative bounce on one body of -5.0 and the other one +5.5 and they should amount to +0.5 and be within bounds... But why? We don't cater to undefined use cases, and if there's no implemented physics engine that actually *supports* values outside 0..1 (as in, it's designed in a way where those values actually make sense, not that they can incidentally produce an outcome which is not outright a bug), there's no reason to future proof.

If you have a very non-standard use case, you can use a script to set those values to arbitrary values. Or create a UnlimitedPhysicsMaterial resource in a script :)

We can discuss this in https://github.com/godotengine/godot-proposals, but I don't see the point in allowing values that don't make sense currently in the engine.

Reply:
Thanks, I'm changing CLAMP to 0, 10 in my fork...

Reply:
As mentioned, if you have a use case for unclamped bounce or friction, please open a proposal on https://github.com/godotengine/godot-proposals

This PR did not restrict anything further than what the codebase already does - even if you could input 10 before this change in the inspector, the engine would clamp it to 1 in `body_pair_2d_sw.cpp` as I've demonstrated above.